src/pybind/mgr/pg_autoscaler/module.py

```diff
@@ -527,6 +527,13 @@ def _calc_final_pg_target(
         ))
         return final_ratio, pool_pg_target, final_pg_target
 
+    def get_dynamic_threshold(final_pg_num: int, default_threshold: float) -> float:
+        if abs(final_pg_num - 512) <= 64 or abs(final_pg_num - 1024) <= 128:
+            return 1.0
+        elif abs(final_pg_num - 2048) <= 256:
+            return 2.0
+        return default_threshold
+
```

Comment:
Not gonna lie, the math is a little complicated, 64, 128 and 256 I guess are marginal difference window.

final_pg_num here will always be nearest power of 2, therefore, can't we just map the exact target to the threshold:

```suggestion
def get_dynamic_threshold(final_pg_num: int, default_threshold: float = 3.0) -> float:
    if final_pg_num in (512, 1024):
        return 1.0
    elif final_pg_num == 2048:
        return 2.0
    return default_threshold

```

Also should we make it such that this dynamic threshold only applies to scaling up the PGs and not scale down? Maybe for scale-down we only resort to default.

Reply:
My sense is that this threshold exists to introduce hysteresis, avoiding flapping when pool usage dips above and below the inflection point.  To that end, might I suggest on line 538 a value of 1.1 instead of 1.0?

Honestly I've been tempted to try to change the existing default threshold to something like 2.1.  My sense is that the clusters that suffer most from the default value of 3.0 are the ones that are small to medium in size, so this would benefit them.

And of course I would raise the target to 250 but that's a different discussion and an existing PR.

Reply:
@anthonyeleven Okay I might be missing something here, but I don't see the benefit of having 1.1 threshold.
The final_pg_num will always be the next power of two, e.g.,
```
256 → 512: 256×1.1 = 281.6 < 512 --> scale up (same as 1.0)
```

The only case I can see that this bump will help is probably when we are no longer quantizing the final_pg_num to power of twos.